airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -0,0 +1,1152 @@
+      domain.resolvedAddress = resolverEntity.resolvedAddress;
+    }
+    // do recursive domain delete
+    domain.lastBlock = block.id;
+    recurseSubdomainCountDecrement(domain, chainId, block, tokenAddress);
+    domain.save();
+
+    // create new resolver transaction
+    getOrCreateAirDomainNewResolverTransaction(
+      previousResolverId,
+      resolverEntity.address,
+      block,
+      transactionHash,
+      logIndex,
+      domain,
+    )
+  }
+
+  /**
+   * @dev This function tracks a new TTL transaction
+   * @param node domain node
+   * @param newTTL new TTL
+   * @param chainId chain id
+   * @param blockHeight block number
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param logIndex event log index
+   * @param transactionHash transaction hash
+   * @param tokenAddress contract address of nft token
+   * @param fromOldRegistry specifies if the event is from the old registry
+   */
+  export function trackDomainNewTTLTransaction(
+    node: string,
+    newTTL: BigInt,
+    chainId: string,
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    logIndex: BigInt,
+    transactionHash: string,
+    tokenAddress: string,
+    fromOldRegistry: boolean,
+  ): void {
+    // get block
+    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
+    // get domain
+    let domain = getOrCreateAirDomain(new Domain(node, chainId, block, tokenAddress));
+    if (fromOldRegistry && domain.isMigrated == true) {
+      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
+      return;
+    }
+    // get previous ttl
+    let oldTTL: BigInt | null = null;
+    if (domain.ttl) {
+      oldTTL = domain.ttl;
+    }
+    // update domain ttl
+    domain.ttl = newTTL;
+    domain.lastBlock = block.id;
+    domain.save();
+    // create AirDomainNewTTLTransaction
+    getOrCreateAirDomainNewTTLTransaction(
+      transactionHash,
+      block.number,
+      logIndex,
+      oldTTL,
+      newTTL,
+      block,
+      domain,
+    )
+  }
+
+  /**
+   * @dev This function tracks a new name registered transaction
+   * @param transactionHash transaction hash
+   * @param blockHeight block number
+   * @param blockHash block hash
+   * @param blockTimestamp block timestamp
+   * @param logIndex event log index
+   * @param chainId chain id
+   * @param registrantAddress registrant address
+   * @param expiryTimestamp domain expiry date
+   * @param cost domain registration cost
+   * @param paymentToken payment token address - can be null
+   * @param labelId label id
+   * @param rootNode root node byte array
+   * @param tokenAddress contract address of nft token
+   */
+  export function trackNameRegisteredTransaction(
+    transactionHash: string,
+    blockHeight: BigInt,
+    blockHash: string,
+    blockTimestamp: BigInt,
+    logIndex: BigInt,
+    chainId: string,
+    registrantAddress: string,
```

Comment:
that is because owner changes twice in while registering a domain, once its from 0x to contract and then from contract to actual owner. That is why this is being handled in the domain owner changed handler.